libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -32,6 +32,52 @@ using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+void copyMissingTags(StructurallyDocumentedAnnotation& _target, CallableDeclarationAnnotation const& _source)
+{
+	auto& sourceDoc = dynamic_cast<StructurallyDocumentedAnnotation const&>(_source);
+
+	set<string> existingTags;
+
+	for (auto& iterator: _target.docTags)
+		existingTags.insert(iterator.first);
+
+	for (auto& [tag, content]: sourceDoc.docTags)
+		if (tag != "inheritdoc" && !existingTags.count(tag))
+			_target.docTags.emplace(tag, content);
+}
+
+CallableDeclaration const* findBaseCallable(std::set<CallableDeclaration const*>& _baseFuncs, string const& _name)
+{
+	for (CallableDeclaration const* baseFuncCandidate: _baseFuncs)
+		if (baseFuncCandidate->annotation().contract->name() == _name)
+			return baseFuncCandidate;
+		else if (auto callable = findBaseCallable(baseFuncCandidate->annotation().baseFunctions, _name))
+			return callable;
+
+	return nullptr;
+}
+
+bool parameterNamesEqual(CallableDeclaration const& _a, CallableDeclaration const& _b)
+{
+	std::vector<ASTPointer<VariableDeclaration>> const& aParams = _a.parameters();
+	std::vector<ASTPointer<VariableDeclaration>> const& bParams = _b.parameters();
+
+	return equal(
```

Comment:
Try using boost::ranges::equal(...) here, for example like:
```suggestion
	return boost::range::equal(_a.parameters(), _b.parameters(), [](auto const& pa, auto const& pb) { return pa->name() == pb->name(); };
```

To effectively making this implementation a one-liner.

Reply:
I thought we try to avoid boost :P